balboa-http/src/it/scala/com/socrata/balboa/server/MetricsIntegrationTest.scala

```diff
@@ -2,34 +2,139 @@ package com.socrata.balboa.server
 
 import java.net.URL
 
+import com.socrata.balboa.metrics.Metric.RecordType
+import com.socrata.balboa.metrics.{Metric, Metrics}
+import com.socrata.balboa.metrics.data.{DataStoreFactory}
 import com.stackmob.newman.ApacheHttpClient
 import com.stackmob.newman.dsl.GET
+import com.stackmob.newman.response.HttpResponse
 import com.stackmob.newman.response.HttpResponseCode.{BadRequest, NotFound, Ok}
-import org.json4s.jackson.JsonMethods.parse
-import org.scalatest.{FlatSpec, Matchers}
+import org.json4s.jackson.JsonMethods.{parse, pretty, render}
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+import scala.language.implicitConversions
 
-import scala.concurrent._
+import scala.concurrent.Await
 import scala.language.postfixOps
 
-class MetricsIntegrationTest extends FlatSpec with Matchers {
+class MetricsIntegrationTest extends FlatSpec with Matchers with BeforeAndAfterEach {
   implicit val httpClient = new ApacheHttpClient
+  val ds = DataStoreFactory.get()
+  val testMetName = "testMetricsPersisted"
+  val testEntPrefix = "testMetricsEntity"
+  var testEntName = ""
+
+  class AssertionJSON(j: => String) {
+    def shouldBeJSON(expected: String) = {
+      val actualObj = parse(j)
+      val expectObj = parse(expected)
+
+      withClue(
+        "\nTextual actual:\n\n" + pretty(render(actualObj)) + "\n\n\n" +
+        "Textual expected:\n\n" + pretty(render(expectObj)) + "\n\n")
+        { actualObj should be (expectObj) }
+    }
+  }
+  implicit def convertJSONAssertion(j: => String): AssertionJSON = new AssertionJSON(j)
+
+  def awaitResponse(url: URL): HttpResponse = {
+    Await.result(GET(url).apply, Config.RequestTimeout)
+  }
+
+  // Ensures each test interacts with a unique entity
+  override def beforeEach() = {
+    testEntName = testEntPrefix + "-" + java.util.UUID.randomUUID().toString
+  }
 
   "Retrieve /metrics range with no range" should "be fail" in {
     val url = new URL(Config.Server, "/metrics/fake.name/range")
-    val response = Await.result(GET(url).apply, Config.RequestTimeout)
+    val response = awaitResponse(url)
```

Comment:
`awaitResponse` puts too much emphasis on the threading aspects of the method instead of on its actual function. `httpGet` or `httpGetSync` would more clearly describe what the method does.